lib/dockerregistry/types.go

```diff
@@ -237,6 +245,26 @@ type RegistryContext struct {
 // "gcr.io/google-containers".
 type RegistryName string
 
+// Filestore holds information about a filestore (e.g. GCS or S3 bucket),
+// to be written in a manifest file.
+type Filestore struct {
+	// Base is the leading part of an artifact path, including the scheme.
+	// It is everything that is not the actual file name itself.
+	// e.g. "gs://prod-artifacts/myproject"
+	Base           string `yaml:"base,omitempty"`
+	ServiceAccount string `yaml:"service-account,omitempty"`
+	Src            bool   `yaml:"src,omitempty"`
+}
+
+// File holds information about a file artifact.
+// File artifacts are copied from a source Filestore to N destination Filestores
+type File struct {
+	// Name is the relative path of the file, relative to the Filestore base
+	Name string `yaml:"name"`
+	// SHA256 holds the SHA256 hash of the specified file (hex encoded)
+	SHA256 string `yaml:"sha256,omitempty"`
```

Comment:
I stated this earlier somewhere else but it would be nice to share the same "style" as we have for the image promoter manifest, where we use a string `sha256:xxxx..`, mainly for consistency; then the field name would instead be `Hash` or some other generic field to denote a checksum.